### PR TITLE
16916 Suppress Certificate Dissolution for Admin Dissolution

### PR DESCRIPTION
**Pull request by kzdev420**
*Issue #:* /bcgov/entity#16916

*Description of changes:*

Suppress certificate dissolution pdf for admin dissolution.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the lear license (Apache 2.0).

### CODE REVIEW

File: legal-api/src/legal_api/core/filing.py [hunk 1]
```diff
@@ -520,6 +520,13 @@ def get_document_list(business,  # pylint: disable=too-many-locals disable=too-m
                 adds = [FilingMeta.get_all_outputs(business.legal_type, doc) for doc in legal_filings]
                 additional = set([item for sublist in adds for item in sublist])
 
+                print(filing.filing_type, filing.storage.filing_sub_type, filing.storage.submitter_roles)
```
Comment (vysakh-menon-aot) [inline legal-api/src/legal_api/core/filing.py]: remove print

File: legal-api/src/legal_api/core/filing.py [hunk 2]
```diff
@@ -520,6 +520,12 @@ def get_document_list(business,  # pylint: disable=too-many-locals disable=too-m
                 adds = [FilingMeta.get_all_outputs(business.legal_type, doc) for doc in legal_filings]
                 additional = set([item for sublist in adds for item in sublist])
 
+                # Supress Certificate of Dissolution for Admin Dissolution
+                if filing.filing_type == Filing.FilingTypes.DISSOLUTION.value and \
+                        filing.storage.filing_sub_type == 'administrative' and \
+                        filing.storage.submitter_roles == UserRoles.staff.value:
+                    additional.remove('certificateOfDissolution')
```
Comment (vysakh-menon-aot) [inline legal-api/src/legal_api/core/filing.py:527]: can you move this logic to `alter_outputs` function (line no 529)

Comment (kzdev420) [reply legal-api/src/legal_api/core/filing.py]: I did

**Review by vysakh-menon-aot — APPROVED** [review]

File: legal-api/src/legal_api/core/meta/filing.py [hunk 3]
```diff
@@ -493,6 +493,10 @@ def alter_outputs(filing: FilingStorage, outputs: set):
                     outputs.remove('certifiedMemorandum')
                 if filing.filing_json['filing']['alteration'].get('rulesInResolution') is True:
                     outputs.remove('certifiedRules')
+        elif filing.filing_type == 'dissolution' and filing.filing_sub_type == 'administrative' and \
+                filing.submitter_roles == 'staff':
```
Comment (argush3) [inline legal-api/src/legal_api/core/meta/filing.py:497]: I don't think the check for `filing.submitter_roles == 'staff'` is required?

Comment (kzdev420) [reply legal-api/src/legal_api/core/meta/filing.py:497]: I think it's required - this is task description ` When filing an Admin Dissolution, which is a Staff filing, stop generating a Certificate of Dissolution.`

Comment (argush3) [reply legal-api/src/legal_api/core/meta/filing.py:497]: The goal of this ticket is to suppress certificate of dissolution for staff and general users for admin dissolution filings right?  To identify a admin dissolution, you only need to use filing type and filing sub-type.

Comment (kzdev420) [reply legal-api/src/legal_api/core/meta/filing.py]: I did

**Review by argush3 — APPROVED** [review]

**Pull request merged**